tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Test_TimeDerivativeTerms.cpp

```diff
@@ -0,0 +1,233 @@
+        make_not_null(&get<::Tags::dt<EvolvedTags>>(*dt_vars_ptr))...,
+        make_not_null(&get<FluxTags>(*volume_fluxes))...,
+        make_not_null(&get<TempTags>(*temporaries))...,
+        get<GradientTags>(partial_derivs)...,
+        tuples::get<ArgTags>(time_derivative_args)...);
+  }
+};
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.GhValenciaDivClean.TimeDerivativeTerms",
+    "[Unit][Evolution]") {
+  using equation_of_state_type = EquationsOfState::IdealFluid<true>;
+  using gh_variables_tags =
+      typename GeneralizedHarmonic::System<3_st>::variables_tag::tags_list;
+  using valencia_variables_tags = typename grmhd::ValenciaDivClean::System<
+      equation_of_state_type>::variables_tag::tags_list;
+
+  using gh_dt_variables_tags = grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+      equation_of_state_type>::gh_dt_tags;
+  using valencia_dt_variables_tags =
+      grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+          equation_of_state_type>::valencia_dt_tags;
+  using dt_variables_type =
+      Variables<tmpl::append<gh_dt_variables_tags, valencia_dt_variables_tags>>;
+
+  using gh_flux_tags = tmpl::list<>;
+  using valencia_flux_tags = grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+      equation_of_state_type>::valencia_flux_tags;
+  using flux_variables_type =
+      Variables<tmpl::append<gh_flux_tags, valencia_flux_tags>>;
+
+  using gh_temp_tags = grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+      equation_of_state_type>::gh_temp_tags;
+  using valencia_temp_tags = grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+      equation_of_state_type>::valencia_temp_tags;
+  using temp_variables_type =
+      Variables<typename grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+          equation_of_state_type>::temporary_tags>;
+
+  using gh_gradient_tags = tmpl::transform<
+      grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+          equation_of_state_type>::gh_gradient_tags,
+      tmpl::bind<::Tags::deriv, tmpl::_1, tmpl::pin<tmpl::size_t<3_st>>,
+                 tmpl::pin<Frame::Inertial>>>;
+  using valencia_gradient_tags = tmpl::list<>;
+  using gradient_variables_type = Variables<gh_gradient_tags>;
+
+  using gh_arg_tags = grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+      equation_of_state_type>::gh_arg_tags;
+  using valencia_arg_tags = grmhd::GhValenciaDivClean::TimeDerivativeTerms<
+      equation_of_state_type>::valencia_arg_tags;
+  using all_valencia_arg_tags =
+      typename grmhd::ValenciaDivClean::TimeDerivativeTerms::argument_tags;
+  using arg_variables_type = tuples::tagged_tuple_from_typelist<
+      tmpl::append<gh_arg_tags, valencia_arg_tags>>;
+
+  const size_t element_size = 10_st;
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> dist(0.1, 1.0);
+
+  dt_variables_type expected_dt_variables{element_size};
+  dt_variables_type dt_variables{element_size};
+  // the dt_tilde_d isn't set by the valencia div clean system, so we
```

Comment:
if you want, you could add `Because the tilde_d evolution equation has no source terms, dt_tilde_d isn't set by...` The flux contributions are handled separately in a different function to keep the functions as method-agnostic as possible :)